### PR TITLE
Update arm gcc toolchain to 11.2-2022.02

### PR DESCRIPTION
**Pull request by FintasticMan**
Updates the buildAndProgram doc and the docker image, but not GitHub Actions.

Once GH Actions use Docker to build the firmware, that should be fine.

### CODE REVIEW

**Comment by Tiggilyboo** [comment]
Works well for me, cheers!

**Review by Riksu9000 — COMMENTED** [review]
The old compiler version is still referenced in some places and should probably be updated.

**Comment by FintasticMan** [comment]
I've just replaced the rest of the references to the old gcc version, except for in the vscode devcontainer because that needs a lot more changes to make it work with the current code. I do think that @geekbozu is working on making it work with the main docker image.

**Comment by geekbozu** [comment]
I am and I'm making it use the main docker file/image so this helps me as well! It's almost done I'm just having some line ending issues that I'm blaming on windows :p

File: doc/buildAndProgram.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Build
 ## Dependencies
 To build this project, you'll need:
- - A cross-compiler : [ARM-GCC (9-2020-q2-update)](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads/9-2020-q2-update)
+ - A cross-compiler : [ARM-GCC (11.2-2022.02)](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/downloads)
```
Comment (NeroBurner) [inline doc/buildAndProgram.md:4]: I think it should be more explicit here that we need a compiler with the arm 32bit eabi target. Maybe add the explicit link for linux amd64 host

```suggestion
 - A cross-compiler : [ARM-GCC eabi (11.2-2022.02)](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/downloads)
   - For an amd64 linux host [gcc-arm-11.2-2022.02-x86_64-arm-none-eabi.tar.xz](https://developer.arm.com/-/media/Files/downloads/gnu/11.2-2022.02/binrel/gcc-arm-11.2-2022.02-x86_64-arm-none-eabi.tar.xz)
```

**Review by NeroBurner — CHANGES_REQUESTED** [review]

**Review by NeroBurner — APPROVED** [review]

**Comment by JF002** [comment]
@FintasticMan Thanks for this PR! Upgrading to an up-to-date version of the compiler is certainly a good thing!
Have you done some tests on builds built by the new version of GCC to confirm that it's working correctly?

**Comment by NeroBurner** [comment]
I've just flashed a build using the mentioned gcc 11 compiler on my pinetime. Everything seems to be fine. What I've tested so far:
- I can play pong
- I can play Twos
- Accellerometer values are shown
- Stopwatch, Alarm and Timer work
- Pinetime Watchface is showing the time and the current bluetooth connection
- Firmware can be validated
- Brightness can be changed in QuickSettings
- The Torch app shines bright

I haven't checked notifications

**Comment by FintasticMan** [comment]
All of my builds for testing features etc. for the past couple of months have all been built with this new compiler version and I haven't noticed any issues or had any crashes. I haven't done any rigorous testing however. To add to the comment above, I have definitely recieved notifications during this time.

**Comment by JF002** [comment]
@FintasticMan 

> All of my builds for testing features etc. for the past couple of months have all been built with this new compiler version and I haven't noticed any issues or had any crashes. I haven't done any rigorous testing however. To add to the comment above, I have definitely recieved notifications during this time.

Great, thanks for this confirmation!

**Review by Riksu9000 — APPROVED** [review]

**Pull request merged**